ee/tables/execparsers/data_table/parser.go

```diff
@@ -0,0 +1,92 @@
+package data_table
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+// data_table is a general parser for an input of data which conforms to columns and rows styled output.
+// Parser options
+// skipLines - The number of initial lines of data to skip. By default no lines are skipped. This can be useful if consistent undesired output/garbage is printed before the data to parse.
+// headers - The set of headers. If left blank, the parser assumes the headers are in the first line of data and splits the line to set them.
+// delimiter - The splitting string. If left blank, the parser assumes the delimiter is whitespace and uses `strings.Fields()` split method.
+type parser struct {
+	skipLines uint
+	headers   []string
+	delimiter string
+}
+
+func Parser(skipLines uint, headers []string, delimiter string) parser {
```

Comment:
If this grows more options, I'd push you to functional arguments, but this is probably fine here

Reply:
Mmm, yeah I'm not sure why I didn't look at doing that initially